internal/controllers/opentelemetrycollector_controller.go

```diff
@@ -285,7 +285,7 @@ func (r *OpenTelemetryCollectorReconciler) Reconcile(ctx context.Context, req ct
 	}
 
 	// Add finalizer for this CR
-	if !controllerutil.ContainsFinalizer(&instance, collectorFinalizer) {
+	if params.Config.CreateRBACPermissions == rbac.Available && !controllerutil.ContainsFinalizer(&instance, collectorFinalizer) {
 		if controllerutil.AddFinalizer(&instance, collectorFinalizer) {
```

Comment:
I'd prefer both this check and `AddFinalizer` to be rolled into a single `maybeAddFinalizer` function.

It'd also be nice to have a unit test for this behavior.